sdk/identity/azure-identity/azure/identity/aio/_credentials/managed_identity.py

```diff
@@ -37,6 +39,15 @@ def __new__(cls, *args, **kwargs):
     def __init__(self, **kwargs: "Any") -> None:
         pass
 
+    async def __aenter__(self):
```

Comment:
This is getting a bit more odd the more methods you add to the class. I know I asked before, but I can't remember the answer - why is this a class with a `__new__` and not just a function that returns the correct credential type?

Reply:
To hide the implementations while providing the right surface for documentation and code completion. It could instead hold an instance of the appropriate credential and wrap its methods... #9302